### PR TITLE
Add related counts for reviewables

### PR DESCRIPTION
**Pull request by chrisseto**
```python
{
    "data": {
        "relationships": {
            "preprints": {
                "links": {
                    "related": {
                        "href": "http://localhost:8000/v2/preprint_providers/psyarxiv/preprints/",
                        "meta": {
                            "accepted": 0,
                            "initial": 1,
                            "pending": 1,
                            "rejected": 0
                        }
                    }
                }
            }
        }
    } 
}
```

### CODE REVIEW

File: reviews/serializers.py [hunk 1]
```diff
@@ -41,6 +41,32 @@ def __repr__(self):
...
+        # Only include counts on detail routes
+        is_detail = not isinstance(self.context['view'], generics.ListAPIView)
+        # Weird hack to avoid being called twice
+        is_related_meta = metadata.get('include_status_counts')
+
+        if show_counts and is_detail and is_related_meta:
+            # Finally, require users to have view_review_logs permissions
+            auth = utils.get_user_auth(self.context['request'])
+            if auth and auth.logged_in and auth.user.has_perm('view_review_logs', provider):
+                metadata = provider.get_reviewable_status_counts()
```
Comment (laurenbarker) [inline reviews/serializers.py]: Will this overwrite existing metadata?

File: reviews/serializers.py [hunk 2]
```diff
@@ -41,6 +41,32 @@ def __repr__(self):
...
+
+    def get_meta_information(self, metadata, provider):
+        metadata = metadata or {}
+
+        # Make counts opt in
+        show_counts = utils.is_truthy(self.context['request'].query_params.get('related_counts', False))
+        # Only include counts on detail routes
+        is_detail = not isinstance(self.context['view'], generics.ListAPIView)
+        # Weird hack to avoid being called twice
+        is_related_meta = metadata.get('include_status_counts')
```
Comment (laurenbarker) [inline reviews/serializers.py]: What causes it to be called twice? Shouldn't `include_status_counts` always be `True` here?

File: reviews/serializers.py [hunk 3]
```diff
@@ -41,6 +41,32 @@ def __repr__(self):
     return IfCanViewLogCreatorField
 
 
+class ReviewableCountingRelationshipField(RelationshipField):
```
Comment (laurenbarker) [inline reviews/serializers.py]: Are there conventions for DRF serializer names?

**Review by laurenbarker — COMMENTED** [review]
Just a few questions. Otherwise looks good to me 🥞

File: reviews/serializers.py (hunk 2 above)
Comment (chrisseto) [reply reviews/serializers.py]: One would think so, but [one](https://github.com/CenterForOpenScience/osf.io/blob/develop/api/base/serializers.py#L716) would be [wrong](https://github.com/CenterForOpenScience/osf.io/blob/develop/api/base/serializers.py#L718).
`include_status_counts` only exists when called with `related_metadata`

File: reviews/serializers.py (hunk 3 above)
Comment (chrisseto) [reply reviews/serializers.py]: Not that I know of, save ending them with `Field`. I'm open to suggestions

File: reviews/serializers.py (hunk 1 above)
Comment (chrisseto) [reply reviews/serializers.py]: It will but there currently isn't any. I'll update it.

File: reviews/serializers.py (hunk 3 above)
Comment (laurenbarker) [reply reviews/serializers.py]: Having a verb in the name seems odd to me -- maybe `ReviewableCountsRelationshipField` (if it's actually weird).

File: reviews/serializers.py (hunk 2 above)
Comment (laurenbarker) [reply reviews/serializers.py]: Oh, so it is still called twice but the counts are only added when called with `related_metadata`?

Comment (chrisseto) [reply reviews/serializers.py]: Bingo, I'll add more detail to the comment

File: reviews/serializers.py (hunk 3 above)
Comment (chrisseto) [reply reviews/serializers.py]: It's very Java -esque, which is a bit odd in python I think `ReviewableCountsRelationshipField` makes more sense as a field name.

**Comment by chrisseto** [comment]
@laurenbarker updated.

**Pull request merged**